src/Gizmos/gizmo.ts

```diff
@@ -161,7 +163,33 @@ export class Gizmo implements IDisposable {
         if (!this._attachedNode) {
             return;
         }
-        if (this._attachedNode.getClassName() === "Mesh" || this._attachedNode.getClassName() === "AbstractMesh" || this._attachedNode.getClassName() === "TransformNode") {
+        var inheritsTargetCamera = this._attachedNode.getClassName() === "FreeCamera"
+                        || this._attachedNode.getClassName() === "FlyCamera"
+                        || this._attachedNode.getClassName() === "ArcFollowCamera"
+                        || this._attachedNode.getClassName() === "TargetCamera"
+                        || this._attachedNode.getClassName() === "TouchCamera"
+                        || this._attachedNode.getClassName() === "UniversalCamera";
+        if  (this._attachedNode.getClassName() === "ArcRotateCamera" || inheritsTargetCamera) {
+            var camera = this._attachedNode as Camera;
+            var transformQuaternion = new Quaternion(0, 0, 0, 1);
+            var position = new Vector3(0, 0, 0);
+            var scaling = new Vector3(1, 1, 1);
+            if (camera.parent) {
+                var parentInv = new Matrix();
+                var localMat = new Matrix();
+                camera.parent.getWorldMatrix().invertToRef(parentInv);
+                this._attachedNode._worldMatrix.multiplyToRef(parentInv, localMat);
+                localMat.decompose(scaling, transformQuaternion, position);
+            } else {
+                this._attachedNode._worldMatrix.decompose(scaling, transformQuaternion, position);
+            }
+            camera.position = position;
+            if (inheritsTargetCamera) {
+                var targetCamera = this._attachedNode as TargetCamera;
+                var angles = transformQuaternion.toEulerAngles();

```

Comment:
a target camera can have a quaternion as rotation. If it is set, copy from the transforrmQuaternion